lib/cloud_pub_sub.ex

```diff
@@ -34,6 +34,8 @@ defmodule CloudPubSub do
   defdelegate connected?(), to: CloudPubSub.Adapter
 
   defdelegate publish(topic, payload, opts), to: CloudPubSub.Adapter
+  defdelegate publish_cast(topic, payload, opts), to: CloudPubSub.Adapter
```

Comment:
Rather than adding a completely new function for this behavior, what about an `async: true` keyword that's captured in the opts list?

```ex
:ok = publish("device/1", "ping", no_reply: true)
```

(I was tempted to use `async: true`, but that sounded confusing because an MQTT publish is async in nature.)

Reply:
We could, but opts is passed all the way down to tortoise so I didnt want to mistakenly use a key that was reserved in that library. I suppose I could look through the library or pop the flag, but then if tortoise does have utilize that key we need to change this code.

Reply:
Something else crossed my mind: Is there any reason not to change the behavior of the original `publish` function to be async (a cast)? I personally kind of expected a publish to be fire-and-forget when I used this library.

Reply:
Other than my fear of breaking anyone's code that is expecting a response returned? No. 
Fire and forget is the default behavior of publish in the Tortoise library, but now that this project has been using publish_sync for so long, Id be worried about people depending on that response code.

Reply:
Also, the only reason I continue to use "publish_sync" in this, is because I dont want random unhandled publish responses coming back to my mailbox later. Publish_sync will return the response to the caller and move on (always :ok for qos < 1), where as publish will send a :info to the calling process at some later time and thats annoying when Im trying to "fire and forget"

Reply:
>Other than my fear of breaking anyone's code that is expecting a response returned? No.

Maybe we should start versioning the library?

> Also, the only reason I continue to use "publish_sync" in this, is because I dont want random unhandled publish responses coming back to my mailbox later.

Could you elaborate on this? Do you need more than a 1-liner like the following to suppress the info message? I'm not making the connection of why the library needs to suppress the info messages rather than letting an integrating developer ignore them in their project if they don't care about them.

```ex
def handle_info({:publish, _}, state), do: {:noreply, state}
```